dbms/src/Storages/StorageDisaggregated.cpp

```diff
@@ -55,7 +64,10 @@ BlockInputStreams StorageDisaggregated::read(
 
     DAGPipeline pipeline;
     buildReceiverStreams(dispatch_reqs, num_streams, pipeline);
-    filterConditions(pipeline);
+
+    NamesAndTypes source_columns = genNamesAndTypesForExchangeReceiver(table_scan);
```

Comment:
Why move this out from `filterConditions`?

Reply:
because we need to reuse `filterConditions` in here 
https://github.com/pingcap/tiflash/blob/4a62a4ac7559e77efa371c062c10f8a01ac1d956/dbms/src/Storages/StorageDisaggregatedRemote.cpp#L118-L119

Reply:
`exchange_receiver` is not inited in StorageDisaggregatedRemote functions